iron-component-page.html

```diff
@@ -8,127 +8,438 @@
+      @apply(--layout-fit);
+    }
 
-    </div>
+    #demo {
+      border: 0;
+      background: transparent;
+      width: 100%;
+      height: 100%;
+      overflow-x: none;
+      overflow-y: auto;
+    }
 
-  </template>
+    #view > * {
+      display: none;
+    }
+
+    #view > .iron-selected {
+      display: block;
+    }
+
+    #docs { padding: 20px; }
 
+    #active {
+      font-size: 20px;
+      font-family: Roboto, Noto;
+      border: 0;
+      background: transparent;
+    }
+
+    paper-toolbar a {
+      font-size: 14px;
+      text-transform: uppercase;
+      cursor: pointer;
+    }
 
+    #cart-icon {
+      margin-left: 10px;
+      cursor: pointer;
+    }
+
+    [catalog-only] {
+      display: none;
+    }
+
+    :host([_catalog]) [catalog-only] {
+      display: block;
+    }
+  </style>
 
+  <template>
+    <hydrolysis-analyzer id="analyzer" src="[[_srcUrl]]" transitive="[[transitive]]" clean analyzer="{{_analyzer}}" loading="{{_loading}}"></hydrolysis-analyzer>
+    <paper-header-panel mode="waterfall">
+      <paper-toolbar>
+        <div class="flex">
+          <!-- TODO: Replace with paper-dropdown-menu when available -->
+          <select id="active" value="{{active::change}}">
+            <template is="dom-repeat" items="[[_elements]]">
+              <option value="[[item.is]]">[[item.is]]</option>
+            </template>
+            <template is="dom-repeat" items="[[_behaviors]]">
+              <option value="[[item.is]]">[[item.is]]</option>
+            </template>
+          </select>
+        </div>
+        <div><a on-tap="_cartAdd" catalog-only><iron-icon icon="add"></iron-icon> Add to Cart</a></div>
+        <iron-selector attr-for-selected="view" selected="{{view}}" id="links" hidden$="[[!_firstDemoUrl]]">
+          <a view="docs"><iron-icon icon="description"></iron-icon> Docs</a>
+          <a view="demo"><iron-icon icon="visibility"></iron-icon> Demo</a>
+        </iron-selector>
+        <iron-icon id="cart-icon" icon="shopping-cart" on-tap="_cartOpen" catalog-only></iron-icon>
+      </paper-toolbar>
+      <div id="content">
+        <iron-selector id="view" selected="[[view]]" attr-for-selected="id">
+          <div id="docs">
+            <iron-doc-viewer descriptor="{{_activeDescriptor}}"></iron-doc-viewer>
+          </div>
+          <iframe id="demo" src="[[_frameSrc(_firstDemoUrl)]]"></iframe>
+        </iron-selector>
+      </div>
+    </paper-header-panel>
+  </template>
 </dom-module>
+
 <script>
+(function() {
+  // var hydrolysis = require('hydrolysis');
+
+  /**
+   * @param {string} url
+   * @return {string} `url` stripped of a file name, if one is present. This
+   *     considers URLs like "example.com/foo" to already be a base (no `.` is)
+   *     present in the final path part).
+   */
+  function _baseUrl(url) {
+    return url.match(/^(.*?)\/?([^\/]+\.[^\/]+)?$/)[1] + '/';
+  }
+
   Polymer({
     is: 'iron-component-page',
-
+    enableCustomStyleProperties: true,
```

Comment:
In general? You can just use `@apply` and friends without declaring you're doing so now? Cool if so.